drivers/scheduler/dcos/dcos.go

```diff
@@ -106,9 +106,43 @@ func (d *dcos) IsNodeReady(n node.Node) error {
 	return nil
 }
 
-func (d *dcos) GetNodesForApp(*scheduler.Context) ([]node.Node, error) {
-	// TODO: Implement this method
-	return nil, nil
+func (d *dcos) GetNodesForApp(ctx *scheduler.Context) ([]node.Node, error) {
+	var tasks []marathon.Task
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*marathon.Application); ok {
+			appTasks, err := MarathonClient().GetApplicationTasks(obj.ID)
+			if err != nil {
+				return nil, &scheduler.ErrFailedToGetNodesForApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get tasks for application %v. %v", obj.ID, err),
+				}
+			}
+			tasks = append(tasks, appTasks...)
+		}
+	}
+
+	nodeMap := make(map[string]node.Node)
+	for _, n := range node.GetNodes() {
```

Comment:
Instead of doing this, would it make sense to have a GetNodeByName() and just use that in the loop below?

Reply:
+1

Reply:
done